person.rb

```diff
@@ -19,10 +26,39 @@ def can_use_services?
     end
   end
 
+  def correct_name
+    @name
+  end
+
   private
 
   def is_of_age
     @age >= 18
   end
   # rubocop:enable Naming/PredicateName
 end
+
+class BaseDecorator < Namable
+  attr_accessor :namable
+
+  def initialize(namable)
+    super()
+    @namable = namable
+  end
+
+  def correct_name
+    @namable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @namable.correct_name.upcase
+  end
```

Comment:
- [ ] This method is meant to capitalize the first letter of the given name, however, in your own case, it is returning all the letters in uppercase. Ruby has a method for [capitalizing](https://www.geeksforgeeks.org/ruby-string-capitalize-method/) a given word. Kindly revisit this method and make it capitalize the first letter of the given name.